### PR TITLE
Move `swift-parser-cli` to its own package

### PR DESCRIPTION
**Pull request by StevenWong12**
This should remove `swift-syntax`'s dependency on `swift-argument-parser`.🥳

I'll add a REAME and sync the changes in the build script later.

### CODE REVIEW

**Comment by kimdv** [comment]
@swift-ci please test

**Review by kimdv — APPROVED** [review]

File: SwiftParserCLI/Package.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// swift-tools-version:5.7
+import PackageDescription
```
Comment (ahoppen) [inline SwiftParserCLI/Package.swift:2]: I would add a newline between the tools version and the `import` to match the style of the top level Package.swift

**Review by ahoppen — COMMENTED** [review]
Thank you. I think some docc articles also need to be updated to include instructions of how to build and run `swift-parser-cli` now. Could you do that as well?

**Review by kimdv — APPROVED** [review]
Would it make sense to make the CI build to ensure it works?

**Review by ahoppen — APPROVED** [review]
Thank you for removing the dependency of swift-syntax on swift-argument-parser

**Comment by ahoppen** [comment]
@StevenWong12 Looks like there’s a merge conflict. Could you resolve it?

**Comment by StevenWong12** [comment]
I've rebased the code.

**Comment by kimdv** [comment]
@swift-ci please test windows

**Comment by StevenWong12** [comment]
Not quite sure why the build failed.
> error: no target named 'swift-parser-cli'
Error: Command failed with non-zero exit code 1:
Command: /Users/ec2-user/jenkins/workspace/swift-syntax-PR-macOS/branch-main/build/buildbot_incremental/toolchain-macosx-x86_64/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift build --package-path /Users/ec2-user/jenkins/workspace/swift-syntax-PR-macOS/branch-main/swift-syntax/SwiftParserCLI --scratch-path /Users/ec2-user/jenkins/workspace/swift-syntax-PR-macOS/branch-main/build/buildbot_incremental/unified-swiftpm-build-macosx-x86_64 --configuration release --multiroot-data-file /Users/ec2-user/jenkins/workspace/swift-syntax-PR-macOS/branch-main/swift/utils/build_swift/resources/SwiftPM-Unified-Build.xcworkspace --verbose --target swift-parser-cli

The same command works locally.

**Comment by StevenWong12** [comment]
Ah, interesting! Here is the pr.
https://github.com/apple/swift/pull/68120

**Comment by kimdv** [comment]
https://github.com/apple/swift/pull/68120

@swift-ci please test

**Comment by ahoppen** [comment]
https://github.com/apple/swift/pull/68120

@swift-ci please test

**Comment by StevenWong12** [comment]
Seems the CI was somehow not triggered. Could you trigger that again?

**Comment by kimdv** [comment]
https://github.com/apple/swift/pull/68120

@swift-ci please test

**Comment by kimdv** [comment]
@StevenWong12 It seems that there is some conflicts.
Would you mind to rebase it? 😁

**Comment by StevenWong12** [comment]
@kimdv Thanks for reminding me of this.😄

 I am on a vacation now. I will rebase this PR when I am back(likes a week or two).

**Comment by StevenWong12** [comment]
@kimdv I've got this rebased. 😉

**Comment by ahoppen** [comment]
Thank you for doing the rebase @StevenWong12 🙏🏽 Let’s see what CI has to say

https://github.com/apple/swift/pull/68120
@swift-ci Please test

**Comment by ahoppen** [comment]
https://github.com/apple/swift/pull/68120
@swift-ci Please test macOS

**Comment by ahoppen** [comment]
https://github.com/apple/swift/pull/68120
@swift-ci Please test macOS

**Comment by kimdv** [comment]
@StevenWong12 could you please run swift format?

**Comment by kimdv** [comment]
@swift-ci please test

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by kimdv** [comment]
@swift-ci Please test Windows

**Comment by ahoppen** [comment]
@swift-ci Please test Windows

**Comment by Eric-Catalyst** [comment]
The Editor Extension sample project ([https://github.com/apple/swift-syntax/tree/main/EditorExtension](https://github.com/apple/swift-syntax/tree/main/EditorExtension)) also has a dependency on swift-argument-parser, so this probably should be updated as part of this merge as well.

**Comment by ahoppen** [comment]
https://github.com/apple/swift/pull/68120

@swift-ci Please test

**Comment by ahoppen** [comment]
> The Editor Extension sample project (https://github.com/apple/swift-syntax/tree/main/EditorExtension) also has a dependency on swift-argument-parser, so this probably should be updated as part of this merge as well.

That’s unrelated to this PR but I’m updating the Package.resolved in the editor extension in https://github.com/apple/swift-syntax/pull/2277.

**Comment by ahoppen** [comment]
@swift-ci Please test

**Comment by ahoppen** [comment]
apple/swift#68120

@swift-ci Please test macOS

**Comment by ahoppen** [comment]
apple/swift#68120

@swift-ci Please test Linux

**Pull request merged**